python/lsst_versions/_versions.py

```diff
@@ -196,9 +196,12 @@ def find_lsst_version(repo_dir: str = ".", version_commit: str = "HEAD") -> str:
     else:
         year, week = weekly_name[2:].split(".")
 
-    # Python pre-release versions must can only have a single integer
-    # after the "a".
-    dev_version = f"{relevant_release + 1}.0.0a{int(year):04d}{int(week):02d}{counter:02d}"
+    # Declare the developer version to be an evolution of the current
+    # release but with the year and weak in the minor and patchlevel parts.
+    # Alpha versions for weeklies were used initially but once full releases
+    # are made it becomes very difficult for tooling to ever install the
+    # alphas.
+    dev_version = f"{relevant_release}.{int(year):04d}.{int(week):02d}{counter:02d}"
```

Comment:
```suggestion
    dev_version = f"{relevant_release}.{int(year):4d}.{int(week):2d}{counter:02d}"
```
Is a leading 0 in the year and patch actually meaningful?

Reply:
Now that the week and year are after the "." the "02d" doesn't really matter because the version will be normalized on the next line to remove the leading 0.